packages/marketplace/ui/src/hooks/use-deposit.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from "react";
+import { useAccount, useWalletClient, useWriteContract } from "wagmi";
+import {
+  MARKETPLACE_ADDRESS,
+  MARKETPLACE_ABI,
+  ERC20_TOKEN_ADDRESS,
+  ERC20_TOKEN_ABI,
+} from "@/shared/constants";
+
+export function useDeposit() {
+  const { address } = useAccount();
+  const { data: walletClient } = useWalletClient();
+  const { writeContractAsync, status } = useWriteContract();
+
+  // Estados para trackear la transacción
+  const [isPending, setIsPending] = useState(false);
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [txHash, setTxHash] = useState<string | null>(null);
+
+  useEffect(() => {
+    console.log({ status });
+  }, [status]);
+  const depositFunds = async (orderId: bigint, amount: bigint) => {
+    if (!walletClient) {
+      setIsError(true);
+      setErrorMessage("No wallet connected");
+      return;
+    }
+
+    setIsPending(true);
+    setIsSuccess(false);
+    setIsError(false);
+    setErrorMessage(null);
+    setTxHash(null);
```

Comment:
I think you can delete setIsError, we can compute that from ErrorMessage !== null, and reduce state. 

txHash I think is enough to return on function